src/app.rs

```diff
@@ -571,13 +570,15 @@ pub fn init_logging(color: bool, format: LogFormat, log_level: &str, rate: u64)
 ///
 /// The returned [Provider] must stay in scope for the entire lifetime of the application, as it
 /// will be unloaded when it is dropped.
-pub fn load_openssl_legacy_provider() -> Option<Provider> {
+pub fn load_openssl_legacy_providers() -> Vec<Provider> {
     warn!(message = "DEPRECATED The openssl legacy provider provides algorithms and key sizes no longer recommended for use.");
-    Provider::try_load(None, "legacy", true)
-        .map(|provider| {
-            info!(message = "Loaded openssl legacy provider.");
-            provider
-        })
-        .map_err(|error| error!(message = "Failed to load openssl legacy provider.", %error))
-        .ok()
+    ["legacy", "default"].into_iter().filter_map(|provider_name| {
+        Provider::try_load(None, provider_name, true)
+            .map(|provider| {
+                info!(message = "Loaded openssl provider.", provider = provider_name);
+                provider
+            })
+            .map_err(|error| error!(message = "Failed to load openssl provider.", provider = provider_name, %error))
+            .ok()
```

Comment:
From the tests in my other PR, sometimes a provider can fail.
For example the FIPS provider did not load because it requires a third party library in the system.
So, yes, loading providers can fail. The question is how tolerant Vector should be about it.

If any requested provider fails to load, the user would not get what he/she wanted and therefore maybe is a good idea to make it a hard fail indeed.